### ISSUE TITLE
Issue with map generation

### ISSUE THREAD

**Issue by Kinderschnitzel23** [TurningWheel/Barony#701]
Dear Barony Dev Team,

recently I was playing the quality of death beta V3.8.5, with my friends in online multiplayer. The first floor looked quite strange - the game generated a closed loop. 
![closed loop](https://user-images.githubusercontent.com/121437224/209547594-76d1b113-66d2-4062-a95c-d85ad305fcba.png)

Keep in mind that we found the bug before the V3.8.6 update yesterday, so if this was already fixed that is fantastic. I am reporting this anyways just in case.

Kind regards.

**Comment by SheridanR** [comment]
While odd, I don't consider this a bug. The generator is able to create large inaccessible areas by conscious design. Rarely, this may cause only a very small area of the level to be traversable, as you see here. The only important point is whether or not the exit can be reached from the starting room without any digging or levitation. We did reveal a bug in the generator in this way recently, where the exit would fail to spawn if the starting area was hemmed in a little too much. We have since fixed that issue in 3.8.6.

At any rate, given this design, there's no clear point at which to decide a level is "open enough" for exploration. Should the player always be able to explore at least 30%? 50%? 90%? Should we allow only one room to be inaccessible or several? Why do we think differently about areas accessible only by levitation, and those accessible by digging only? There are many regions of the game such as the underworld which explicitly limit traversal without levitation, besides numerous other secret areas throughout the game.

Others on the team are free to disagree with this opinion but as far as I'm concerned, "inaccessible" areas in the generator have always been a feature, not a bug. Even if it means a level will become completely inaccessible without aid. The game is not ruined just because you miss an opportunity to loot and pillage a single dungeon level, on the contrary it makes an interesting way to set back the player's character's leveling progress.
